python-frontend/src/main/java/org/sonar/python/api/PythonGrammar.java

```diff
@@ -207,7 +207,7 @@ public static void grammar(LexerfulGrammarBuilder b) {
     b.rule(F_STRING_CONTENT).is(b.zeroOrMore(b.firstOf(GenericTokenType.UNKNOWN_CHAR, FORMATTED_EXPR)));
     b.rule(FORMATTED_EXPR).is(
       PythonPunctuator.LCURLYBRACE,
-      EXPR,
+      TEST,
```

Comment:
That's better.
However, we still don't handle [all the possibilities of the grammar](https://docs.python.org/3/reference/lexical_analysis.html#formatted-string-literals).
For example, we still have an FP on unused local var for:
```
def f1():
    x = 42
    print(f"{41,x}")
```
However, that's not explicitly in the scope of SONARPY-815, so I suggest to create another ticket to tackle it later.

Reply:
Indeed, thanks for pointing this out.
I'll reopen SONARPY-726 and mention this in the ticket, as this PR only partly fixes the underlying problem.